autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if g:ale_experimental_multibuffer && has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
You need to use `!` here for the flag. You want the inverse. I think you can safely skip the `has_key` and `>= 0` checks. The API for commands is that they must set a valid buffer number.